sway-parse/src/item/mod.rs

```diff
@@ -144,20 +145,34 @@ impl Parse for FnArg {
 
 impl Parse for FnSignature {
     fn parse(parser: &mut Parser) -> ParseResult<FnSignature> {
+        let visibility = parser.take();
+        let fn_token = parser.parse()?;
+        let name = parser.parse()?;
+        let generics = parser.guarded_parse::<OpenAngleBracketToken, _>()?;
+        let arguments = parser.parse()?;
+        let mut where_clause_opt = None;
+        let return_type = match parser.take() {
+            Some(right_arrow_token) => {
+                let ty = parser.parse()?;
+                where_clause_opt = parser.guarded_parse::<WhereToken, _>()?;
+                FnReturnType::TypedReturn((right_arrow_token, ty))
+            }
+            None => {
+                where_clause_opt = parser.guarded_parse::<WhereToken, _>()?;
+                let Some(open_curly_brace) = parser.peek::<OpenCurlyBraceToken>();
+                FnReturnType::Implicit(ImplicitReturn {
+                    span: open_curly_brace.span(),
+                })
+            }
+        };
```

Comment:
@mohammadfawaz (: making some progress here finally

Reply:
need to add the error handling there, but just excited to finally finish this PR up